### DISCUSSION TITLE
GitHub Branch Protection for preventing users to read a branch

### DISCUSSION THREAD

**Discussion by JalonFlems** [community/community#66431] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I was looking at the github documentation for branch protection and I could not find anything related to preventing a collaborator from reading code on a specific branch. Is it currently not possible to restrict read permissions at the branch level? Is this not available for any of the paid tiers as well? Lastly, will branch protection for read permissions be supported in the near future?

**Comment by LeandroLuna** [comment, marked as answer]
Hi, Jalon!

GitHub's branch protection features primarily focus on write permissions and restrictions, rather than read permissions at the branch level. GitHub allows you to protect branches to prevent certain actions like force pushes, branch deletions, and require code reviews, but it does not provide fine-grained read permissions at the branch level, even for paid tiers.

If you want to restrict collaborators' access to code at a specific branch level, an alternative is to use an external access control system or a private repository configuration with selective invitations.
